applications/concept-catalogue/src/main/kotlin/no/begrepskatalog/conceptregistration/BegreperApiImplK.kt

```diff
@@ -48,30 +51,79 @@ class BegreperApiImplK(val sqlStore: SqlStore) : BegreperApi {
                 }
     }
 
-    override fun setBegrepById(httpServletRequest: HttpServletRequest, @ApiParam(value = "id", required = true) id: String, @ApiParam(value = "", required = true) @Valid @RequestBody begrep: Begrep, @ApiParam(value = "Om begrepet skal valideres") @Valid @RequestParam(value = "validate", required = false) validate: Boolean): ResponseEntity<Begrep> {
-        //We must have an ID that already exists, AND a virksomhet
-        if (!sqlStore.begrepExists(begrep)) {
+    override fun setBegrepById(httpServletRequest: HttpServletRequest?, id: String?, begrep: Begrep?, validate: Boolean?): ResponseEntity<Begrep> {
+        if (id == null) {
+            throw RuntimeException("Attempt to PATCH begrep with no id path variable given")
+        }
+        if (begrep == null) {
+            throw RuntimeException("Attempt to PATCH begrep with no begrep data given. Id provided was $id")
+        }
+
+        if (!sqlStore.begrepExists(id)) {
             throw RuntimeException("Attempt to PUT begrep that does not already exist. Begrep id ${begrep.id}")
         }
+        //Get the begrep, and just update
+        var storedBegrep = sqlStore.getBegrepById(id)
 
-        if (begrep.ansvarligVirksomhet == null) {
-            throw RuntimeException("Attempt to PUT begrep with no accompanying Ansvarlig Virksomhet")
+        if (storedBegrep == null) {
+            throw java.lang.RuntimeException("Stored begrep with id $id was null. This should not happen")
         }
 
-        if (begrep.status == Status.UTKAST) {
-            sqlStore.saveBegrep(begrep)
+        val updatedBegrep = updateBegrep(begrep, storedBegrep)
+
+        if (updatedBegrep.status == Status.UTKAST) {
+            sqlStore.saveBegrep(updatedBegrep)
             return ResponseEntity(HttpStatus.OK)
         } else {
-            if (begrep.anbefaltTerm != null && begrep.definisjon != null) {
-                sqlStore.saveBegrep(begrep)
-                logger.info("Begrep $begrep.id has passed validation for non draft begrep and has been saved ")
-                return ResponseEntity(HttpStatus.OK)
+            if (updatedBegrep.anbefaltTerm != null && updatedBegrep.definisjon != null) {
```

Comment:
Dette ser ut som validering for om begrepet kan publiseres. Kan det være lurt å gjøre validering i egen funksjon hvor alle valideringsregler vedlikeholdes. Da kan du her ha f.eks.:
if (validateBegrep(updatedBegrep) { ...

Reply:
Godt poeng!